dashboard/test/controllers/lti_v1_controller_test.rb

```diff
@@ -431,6 +430,24 @@ def create_valid_jwt_raise_error
     # could confirm more things here
   end
 
+  test 'auth - given a deployment_id not in our system yet, create LtiDeployment' do
+    payload = get_valid_payload
+    jwt = create_jwt_and_stub(payload)
+    post '/lti/v1/authenticate', params: {id_token: jwt, state: @state}
+    deployment = LtiDeployment.find_by(deployment_id: @deployment_id)
+    assert deployment
+    assert_equal deployment, @integration.lti_deployments.first
+  end
+
+  test 'auth - given an existing deployment_id in our system, do not create a new LtiDeployment' do
+    payload = get_valid_payload
+    jwt = create_jwt_and_stub(payload)
+    deployment = LtiDeployment.create(deployment_id: @deployment_id, lti_integration_id: @integration.id)
+    assert deployment
+    post '/lti/v1/authenticate', params: {id_token: jwt, state: @state}
+    assert_equal deployment, @integration.lti_deployments.first
```

Comment:
Minor nitpick, but to more definitively test that it didn't create another deployment, should we also check that the length of `@integration.lti_deployments` is 1?

Reply:
Yes, thanks. This test felt weird, that's a good addition.